server/data/incidentReportingApi.ts

```diff
@@ -423,20 +423,38 @@ export class IncidentReportingApi extends RestClient {
     return convertReportWithDetailsDates(report)
   }
 
-  get staffInvolved(): RelatedObjects<StaffInvolvement> {
+  get staffInvolved(): RelatedObjects<StaffInvolvement, AddStaffInvolvement> {
     return new RelatedObjects(this, 'staff-involved')
   }
 
-  get prisonersInvolved(): RelatedObjects<PrisonerInvolvement> {
+  get prisonersInvolved(): RelatedObjects<PrisonerInvolvement, AddPrisonerInvolvement> {
     return new RelatedObjects(this, 'prisoners-involved')
   }
 
-  get correctionRequests(): RelatedObjects<CorrectionRequest> {
+  get correctionRequests(): RelatedObjects<CorrectionRequest, AddCorrectionRequest> {
     return new RelatedObjects(this, 'correction-requests', convertCorrectionRequestDates)
   }
 }
 
-class RelatedObjects<ResponseType> {
+type AddStaffInvolvement = {
+  staffUsername: string
+  staffRole: StaffRole
+  comment?: string
+}
+
+type AddPrisonerInvolvement = {
```

Comment:
Should these also have the `*Request` suffix?

Reply:
yes good idea. will add to all type. (there are more in subsequent commits)

Reply:
blindly adding a `Request` suffixes has resulted in “AddQuestionWithResponsesRequest” … while it follows the pattern, does it sound too stupid?

Reply:
a bit yeah 😅 - especially as there is a mixture or domain concept and HTTP concept in there 😬